packages/server/graphql/types/User.ts

```diff
@@ -421,19 +423,57 @@ const User = new GraphQLObjectType<any, GQLContext>({
       },
       resolve: async ({id: userId}, {reflectionId}, {dataLoader}) => {
         const retroReflection = await dataLoader.get('retroReflections').load(reflectionId)
-        if (!retroReflection) return []
+        if (!retroReflection) {
+          return standardError(new Error('Invalid reflection id'), {userId})
+        }
         const {meetingId} = retroReflection
         const meetingMemberId = toTeamMemberId(meetingId, userId)
-        const [viewerMeetingMember, reflectionGroups] = await Promise.all([
+        const r = await getRethink()
+        const [viewerMeetingMember, reflections] = await Promise.all([
           dataLoader.get('meetingMembers').load(meetingMemberId),
-          dataLoader.get('retroReflectionGroupsByMeetingId').load(meetingId)
+          r
+            .table('RetroReflection')
+            .getAll(meetingId, {index: 'meetingId'})
+            .filter({isActive: true})
+            .orderBy('createdAt')
+            .run()
         ])
         const {meetingId: viewerMeetingId} = viewerMeetingMember
         if (viewerMeetingId !== meetingId) {
-          standardError(new Error('Not on team'), {userId})
-          return []
+          return standardError(new Error('Not on team'), {userId})
         }
-        return reflectionGroups
+        let currentThreshold: number | null = AUTO_GROUPING_THRESHOLD
+        const similarGroupIds = new Set<string>()
+        const reflectionsCount = reflections.length
+        const maxGroupSize = Math.min(reflectionsCount, 10)
+        while (currentThreshold) {
+          const {groupedReflections, nextThresh} = groupReflections(reflections, {
+            groupingThreshold: currentThreshold,
+            maxGroupSize: reflectionsCount,
+            maxReductionPercent: 1
+          })
+          const spotlightGroup = groupedReflections.find(
+            (group) => group.reflectionId === reflectionId
+          )
+          if (!spotlightGroup) break
+          for (const group of groupedReflections) {
+            if (similarGroupIds.size === maxGroupSize) {
+              currentThreshold = null
+              break
+            }
+            if (
+              group.reflectionGroupId === spotlightGroup.reflectionGroupId &&
+              group.oldReflectionGroupId !== spotlightGroup.oldReflectionGroupId
+            ) {
+              similarGroupIds.add(group.oldReflectionGroupId)
+            }
+            currentThreshold = nextThresh
+          }
+        }
```

Comment:
In the while loop, we're looking through all of the `groupedReflections` rather than the `reflectionGroups`. `spotlightGroup` is the grouped reflection in the source (top bit) of the Spotlight. If we break the loop when we reach 10 `groupedReflections`, there may be less than 10 `reflectionGroups` as several reflections could be in one group. I think we need to keep track of the unique old `reflectionGroupIds` in the loop so that we can guarantee 10 groups.

I agree it's confusing - I've pushed an update changing group to `groupedReflection` to try to add more clarity. Let me know what ya think!